### DISCUSSION TITLE
autofocus cursor on html elements in custom jspsych plugin?

### DISCUSSION THREAD

**Discussion by IrisxBi** [jspsych/jsPsych#1039] [Q&A] [answered]
Hi everyone,

I'm trying to create a custom plug-in that basically displays a prompt with a free-response input box under it each time it is called. I've added an autofocus property to the html input element so that the cursor is placed in the input box automatically (rather than the user needing to click on it), but this only works for the first time the plug-in is called, and not for subsequent trials using the same plugin. 

Does anyone have any pointers on how to fix this? Would really appreciate it! I've included the code for the plugin in a .txt file. 
[jspsych-scrambled-sentence-response.txt](https://github.com/jspsych/jsPsych/files/5198139/jspsych-scrambled-sentence-response.txt)

**Comment by becky-gilbert** [comment, marked as answer]
I think the inline autofocus property is what causes the autofocus to work on the first trial but not subsequent trials. I think you had the right idea about calling the `focus` JavaScript method but I see that you commented it out:
```javascript
//$('demo').focus()
```
That line is using jQuery. Are you using jQuery? 
If not you could use the "vanilla" JavaScript approach. If you give your text input box an ID like "demo" then you can do it like this:
```javascript
document.getElementById("demo").focus();
```
Also it looks like you're missing a > when you set up your [input box HTML](https://www.w3schools.com/tags/att_input_type_text.asp), and you don't need the `</input>`. So it should look like this:
```javascript
html += '<input type="text" class="demo" name="jspsych-scrambled-sentence-response" autofocus="autofocus" size="40">';
```
Not sure if that's causing any problems.

**Reply by IrisxBi** [reply]
Yes, that fixed it! Thanks so much!
